asabiris/output/ms365/service.py

```
@@ -208,11 +713,15 @@ def _post(token):
-		# Retry on 401
+		# Retry on 401 (token expired, consent revoked, etc.)
 		if resp.status_code == 401:
-			L.info("Token expired—retrying")
-			token = self._get_access_token(force_refresh=True)
-			resp = _post(token)
+			L.info("Token expired or invalid—retrying with force_refresh.")
+			try:
+				token = await self._get_access_token_async(force_refresh=True)
+			except ASABIrisError as e:
+				# Refresh failed → bubble up
+				raise e
```

Ruff (0.14.11)

Remove exception handler; error is immediately re-raised

(TRY203)